transformations/replace_with_hyponyms/transformation.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+import spacy
+#from checklist.perturb import Perturb
+from checklist.editor import Editor
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+class ReplaceHyponyms(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+
+    def __init__(self, n=1, seed=0, max_output=2):
+        # TODO: Do not repeat parse computations.
+        super().__init__(seed)
+        self.nlp = spacy.load("en_core_web_sm")
+        self.n = n
+        self.max_output = max_output
+
+    def generate(self, sentence: str):
+        np.random.seed(self.seed)
+        words = []
+        perturbed_texts = []
+        tokens = self.nlp(sentence)
+        for token in tokens:
```

Comment:
There is high probabilty that for a long sentence (which could have mulltiple 'nouns' Ex: "A boy and a girl were sitting under a tree."), everytime only initial nouns are selected for generating transformations. To avoid that, you can shuffle the token list "tokens" before looping, so that every noun will get an equal chance of getting replaced.